src/utils/polygonUtils.h

```diff
@@ -615,6 +615,12 @@ class PolygonUtils
      */
     static double relativeHammingDistance(const Polygons& poly_a, const Polygons& poly_b);
 
+    static void makeCircle(Point mid, coord_t radius, Polygons& result, float a_step = M_PI / 8);
+
+    /*!
+     * Connect all polygons to their holes using zero widths hole channels, so that the polygons and their outlines are connected together
+     */
+    static Polygons connect(const Polygons& input);
```

Comment:
So the result should be guaranteed to be a `Polygon` rather than `Polygons`, right?

Reply:
Incorrect. There will be a `Polygon` for every part returned by `input.splitIntoParts()`.